src/components/SourcesTree.js

```diff
@@ -54,6 +55,15 @@ let SourcesTree = React.createClass({
       return this.setState({ listItems });
     }
 
+    if (nextProps.selectedSource != this.props.selectedSource) {
+      const hLightItems = getDirectories(
+        nextProps.selectedSource.get("url"),
+        this.state.sourceTree
+      );
+
+      return this.setState({ hLightItems });
```

Comment:
@jasonLaster 

I'm not really sure if there is a faster way to do this :( 
The code that we have right now is that we get an itemList of the target source's parents, including itself, by calling `getDirectories`.  For example, if we have "folder/subfolder/targetFile", `getDirectories` gets us a list that contains `["folder", "folder/subfolder", "folder/subfolder/targetFile"]`. 

In `ManagedTree`, we first check if `folder/subfolder/targetFile` is in the expanded, which means if it's visible. If it is, we will just highlight it. If not, we check if `folder` is opened. If it is, we check `folder/subfolder`.  And this process is finding the lowest directory of the target source.